prov/efa/src/rxr/rxr_read.c

```diff
@@ -82,7 +82,47 @@ int rxr_locate_rma_iov_pos(struct fi_rma_iov *rma_iov, int rma_iov_count, size_t
 ssize_t rxr_read_mr_reg(struct rxr_ep *ep, struct rxr_read_entry *read_entry)
 {
 	size_t i;
-	int err;
+	int err, pkt_offset;
+	struct rxr_pkt_entry *pkt_entry;
+	struct rxr_pkt_entry *pkt_entry_copy;
+
+	if (read_entry->x_entry_type == RXR_PKT_ENTRY) {
```

Comment:
This section doesn't seem applicable for a function called `rxr_read_mr_reg`, can this go in its own function?

Reply:
This section is still preparing memory registration, so I think it is appropriate to leave it here. Maybe rename the function to ```rxr_read_prepare_mr```?

Reply:
I'd still split the function even if it has to be called in `rxr_read_mr_reg` for readability.

Reply:
Split into a new function in new revision.